Update conf reg email notification format to support in-person/virtual attendance

Scotchester opened the pull request:
The conference registration notification email template was not updated to support the new option for in-person or virtual attendee registration, and the way in which that affects the capacity. This PR corrects that.

## Changes

- Updates `ConferenceNotifier` so that it has the information it needs about the different attendee counts and the question of being at capacity
- Updates the email body and subject templates to output the information accurately

## Testing

1. With a recent DB dump, edit http://localhost:8000/admin/pages/11796/edit/ to change the capacity to `1`.
1. Submit a registration for a virtual attendee
1. Run `./cfgov/manage.py conference_notify 11796 -t a@b.c --dry-run` to see the resulting email
1. Submit a registration for a physical attendee
1. Run `./cfgov/manage.py conference_notify 11796 -t a@b.c --dry-run` to see the resulting email
1. Submit another registration for a virtual attendee
1. Run `./cfgov/manage.py conference_notify 11796 -t a@b.c --dry-run` to see the resulting email

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

chosak commented on cfgov/data_research/templates/data_research/conference_notify_subject.txt:
```diff
@@ -1 +1 @@
-[{{ conference_name }}] Attendance {{ at_capacity|yesno:"Full,Update" }}
+[{{ conference_name }}] Attendance Update{{ at_capacity|yesno:" (in-person capacity reached)," }}
```
Is there an extra trailing comma after `reached)`?

Scotchester replied:
No, it is the separation between the `yes` output and the `no` output (which should be none). Very strange Django filter, this is.

chosak replied:
Ah, right. Thanks for the correction.

Scotchester commented:
Updated this to move the in-person/virtual attendee filters to a custom QuerySet being used as a Manager on the `ConferenceRegistration` model, at @chosak's suggestion.

Rebased on master, so re-pull it if you already pulled it for testing.

chosak commented on cfgov/data_research/models.py:
```diff
@@ -15,11 +15,43 @@
+
+
 class ConferenceRegistration(models.Model):
+    IN_PERSON = 'In person'
```
To avoid duplication, can we remove [these hardcoded values](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/data_research/forms.py#L36) on the registration form class and instead use these new ones on the model?

Scotchester replied:
Gah, yeah. I meant to do that.
